apps/server/src/modules/account/services/account-idm.service.ts

```diff
@@ -1,118 +1,117 @@
 import { IdentityManagementOauthService, IdentityManagementService } from '@infra/identity-management';
 import { ObjectId } from '@mikro-orm/mongodb';
 import { Injectable, NotImplementedException } from '@nestjs/common';
+import { ConfigService } from '@nestjs/config/dist/config.service';
 import { EntityNotFoundError } from '@shared/common';
 import { IdmAccount, IdmAccountUpdate } from '@shared/domain/interface';
 import { Counted, EntityId } from '@shared/domain/types';
 import { LegacyLogger } from '@src/core/logger';
-import { AccountIdmToDtoMapper } from '../mapper';
-import { AccountLookupService } from './account-lookup.service';
+import { Account, AccountSave } from '../domain';
+import { AccountIdmToDoMapper } from '../repo/mapper';
 import { AbstractAccountService } from './account.service.abstract';
-import { AccountDto, AccountSaveDto } from './dto';
+import { AccountConfig } from '../account-config';
 
 @Injectable()
 export class AccountServiceIdm extends AbstractAccountService {
 	constructor(
 		private readonly identityManager: IdentityManagementService,
-		private readonly accountIdmToDtoMapper: AccountIdmToDtoMapper,
-		private readonly accountLookupService: AccountLookupService,
+		private readonly accountIdmToDoMapper: AccountIdmToDoMapper,
 		private readonly idmOauthService: IdentityManagementOauthService,
-		private readonly logger: LegacyLogger
+		private readonly logger: LegacyLogger,
+		private readonly configService: ConfigService<AccountConfig, true>
 	) {
 		super();
 	}
 
-	async findById(id: EntityId): Promise<AccountDto> {
+	async findById(id: EntityId): Promise<Account> {
 		const result = await this.identityManager.findAccountById(id);
-		const account = this.accountIdmToDtoMapper.mapToDto(result);
+		const account = this.accountIdmToDoMapper.mapToDo(result);
 		return account;
 	}
 
-	async findMultipleByUserId(userIds: EntityId[]): Promise<AccountDto[]> {
+	// TODO: this needs a better solution. probably needs followup meeting to come up with something
+	async findMultipleByUserId(userIds: EntityId[]): Promise<Account[]> {
 		const results = new Array<IdmAccount>();
 		for (const userId of userIds) {
```

Comment:
const promises = userIds.map((userId) -> this.identityManager.findAccountByDbcUserId(userId));

const results = await Promise.all(promises) / Promise.settled(promises);

But keyclock has no batch, find, multi solution to find many users?